File: go/pangram/pangram.go
// Package pangram has a pangram checker
package pangram

import (
	"unicode"
)

// IsPangram returns whether the input is a pangram
func IsPangram(in string) bool {
	chars := make(map[rune]bool)
	for _, v := range in {
		newV := unicode.ToLower(v)
		chars[newV] = true
	}
	for c := 'a'; c <= 'z'; c++ {
		if _, ok := chars[c]; !ok {
			return false
		}
	}
	return true
}

File: cpp/pangram/pangram.cpp
#include "pangram.h"

namespace pangram {

    bool is_pangram(const std::string &input) {
        bool cache[alpha_size] = {false};
        for (const auto &i : input) {
            if (std::isalpha(i)) {
                int ch = int(std::tolower(i) - 'a');
                cache[ch] |= true;
            }
        }
        for (int i = 0; i < alpha_size; i++) {
            if (!cache[i]) return false;
        }
        return true;
    }

}
